### DISCUSSION TITLE
Managing project dependencies

### DISCUSSION THREAD

**Discussion by ThinkTransit** [micropython/micropython#9097] [General]
I'm interested to know how others manage dependencies in their MicroPython projects.

I have been haphazardly copying files from micropython-lib and other repos into various projects, with little regard for version control etc..

Is there a way to manage project dependencies in MicroPython, similar to pipenv for cpython?

**Comment by jimmo** [comment]
This is something we're actively working on!

See #8914 (and related #9037 and https://github.com/micropython/micropython-lib/pull/506).

One of the goals here is that you will have a "project manifest" describing your dependencies, which can either be used to freeze firmware, or to pass to mpremote to deploy to the filesystem.

**Comment by ThinkTransit** [comment]
Wow, I'm going to spend some time reading but looks good at first glance. I will definitely be able to help out with the testing of this.
